aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -126,18 +126,26 @@ module AptosFramework::Coin {
 
     // Public functions
 
-    /// Burn coin with capability.
+    /// Burn `coin` with capability.
     public fun burn<CoinType>(
         account: &signer,
         coin: Coin<CoinType>,
-    ) acquires BurnCapability, CoinInfo {
+    ) acquires CoinInfo {
         let account_addr = Signer::address_of(account);
         assert!(
             exists<BurnCapability<CoinType>>(account_addr),
             Errors::not_published(ENO_BURN_CAPABILITY),
         );
-        let _cap = borrow_global<BurnCapability<CoinType>>(account_addr);
 
+        burn_with_capability(coin, BurnCapability<CoinType> { });
```

Comment:
this should try to acquire the capability from signer? I think we can unify it with the assertion

Reply:
I've changed the approach here so this comment no longer applies. PTAL again 🙏